### PR TITLE
Completed Task 8

### PR DESCRIPTION
**Pull request by JoshuaGoodwill**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -37,6 +37,29 @@ const fetchComments = (id) => {
     });
 };
 
+
```
Comment (JVal-NC) [inline src/utils.js:40]: Can we rename this file api, whilst it is a util, api might be more precise.

File: src/components/NewComment.jsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    posted: 0,
+    message: "",
+  });
+
+  useEffect(() => {
+    fetchUsers()
+      .then(({ users }) => {
+        return users;
+      })
+      .then((users) => {
```
Comment (JVal-NC) [inline src/components/NewComment.jsx:22]: The logic of this then block could be incorporated into the previous FYI.

File: src/components/NewComment.jsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    ) {
+      return;
+    } else {
+      setFormValues((currValues) => {
+        return { ...currValues, [event.target.id]: event.target.value };
+      });
+    }
+  };
+
+  const submitForm = (event) => {
```
Comment (JVal-NC) [inline src/components/NewComment.jsx:47]: We can currently submit a comment twice if we press submit in quick succession.

Also, the new comments appear at the top but appear at the bottom after refreshing.

Good feedback to user on errors but "commented on NaN/NaN/NaN" is a bit buggy, can we give this a different value.

**Review by JVal-NC — COMMENTED** [review]
Josh, strong work so far but a couple bugs to fix.

Another nchelp pr once fixed up please.

File: src/components/CommentsList.jsx [hunk 4]
```diff
@@ -24,19 +25,30 @@ const CommentsList = ({ id }) => {
...
-            return (
-              <li className="commentsListItem" key={comment.comment_id}>
-                <h4>
-                  {comment.author} commented on {fullDate.getDate()}/
-                  {fullDate.getMonth()}/{fullDate.getFullYear()}:
-                </h4>
-                <p>{comment.body}</p>
-              </li>
-            );
+            if (comment.author === "ERROR") {
```
Comment (Augs0) [inline src/components/CommentsList.jsx]: This seems like an unusual condition..why would author be error?

File: src/components/NewComment.jsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            posted: 1,
+            message: "Successfully posted comment.",
+          });
+        })
+        .catch((err) => {
+          console.log(err);
+          setCommentsArr((currComments) => {
+            return [
+              {
+                author: "ERROR",
```
Comment (Augs0) [inline src/components/NewComment.jsx]: I see the connection now. A tad unusual to do it like so. We could simply have an error state and deal with it that way

**Review by Augs0 — CHANGES_REQUESTED** [review]
Almost there Josh just a few things.
- I can't post a blank comment which is great but I get no feedback telling me why I can't post. We need a bit of validation here
- Not sure if it's just my side but the hyperlinks to read article have a low contrast and are hard to read
- The input for body should ideally be text area as it's hard to fit more than a few words in input without scrolling back to see your message

**Pull request merged**